tests/imfile-file-descriptor-exhasution.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+# This file is part of the rsyslog testbench, released  under ASL 2.0
+. ${srcdir:=.}/diag.sh init
+. $srcdir/diag.sh check-inotify
+
+generate_conf
+add_conf '
+# comment out if you need more debug info:
+	global( debug.whitelist="on"
+		debug.files=["imfile.c"])
+
+module(load="../plugins/imfile/.libs/imfile"
+       mode="inotify" normalizePath="off")
+
+input(type="imfile" File="./'$RSYSLOG_DYNNAME'.input.*.log" Tag="file:"
+	Severity="info" Facility="local7")
+
+'
+# generate input files first. Note that rsyslog processes it as
+# soon as it start up (so the file should exist at that point).
+# creating 1024 files to be parsed by imfile
+for  (( i= 0 ; i < 100 ; ++i ))
+do
+	echo test >> $RSYSLOG_DYNNAME.input.$i.log;
+done
+
+# Start rsyslog now
+startup
+
+#keep a boolean to track if FD exhasuted
+
+# keep adding files and check wether the FD exhasuted
+num_of_files=0
+
+for (( i = 100; i < 1200 ; ++i ))
+do
+	echo test >> $RSYSLOG_DYNNAME.input.$i.log;
+	if content_check --check-only "Too many open files" $RSYSLOG_DYNNAME.started; then
+		printf 'Failed to open file $RSYSLOG_DYNNAME.input.%d.log a FD exhasuted\n' "$i"
```

Comment:
now that we are at it again: typo "exhasuted" vs. "exhausted"

Reply:
i will modify this. Thanks for pointing out.